src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1108,6 +1133,45 @@ private static void ParseHeaderNameValue(HttpConnection connection, ReadOnlySpan
             }
         }
 
+        private void ProcessHttp10KeepAliveHeader(HttpResponseMessage response)
+        {
+            if (response.Headers.NonValidated.TryGetValues(KnownHeaders.KeepAlive.Name, out HeaderStringValues keepAliveValues))
+            {
+                string keepAlive = keepAliveValues.ToString();
+                var parsedValues = new UnvalidatedObjectCollection<NameValueHeaderValue>();
+
+                if (NameValueHeaderValue.GetNameValueListLength(keepAlive, 0, ',', parsedValues) == keepAlive.Length)
+                {
+                    foreach (NameValueHeaderValue nameValue in parsedValues)
+                    {
+                        if (string.Equals(nameValue.Name, "timeout", StringComparison.OrdinalIgnoreCase))
+                        {
+                            if (!string.IsNullOrEmpty(nameValue.Value) &&
+                                HeaderUtilities.TryParseInt32(nameValue.Value, out int timeout) &&
+                                timeout >= 0)
+                            {
+                                if (timeout <= 1)
+                                {
+                                    _connectionClose = true;
```

Comment:
Why close it for timeout == 1 second?

Reply:
This is based on the [`TimeoutOffsetSeconds = 1`](https://github.com/dotnet/runtime/blob/2ba292211c3f602040944b4f8c468fa51d6f0046/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L240) constant I used when checking whether we already exceeded the timeout.

If the server requested `timeout == 1`, we would never reuse the connection, so there's no need to keep it around until the next request. (tbh I don't know if this matters at all in practice)

I can use the same constant here and add a comment so it makes more sense.

Reply:
> If the server requested timeout == 1, we would never reuse the connection

Why is that desirable? i.e. why is 1 second any more special than 2 seconds or 10 seconds?

Reply:
> An arbitrary choice here was 1 second for that constant

That seems very arbitrary.  Does this meaningfully help avoid 99% of such races?  Do servers typically send timeout = 1?  My inclination would be to remove the special-casing entirely.

Reply:
The original bug was reported that we were exceeding the timeout - but in fact didn't know about the timeout - and it was suspected that there was a race condition. We probably don't need to have a minimum as the server should have a bit of wiggle room in the value it gives back.

Reply:
It looks like Chrome (and Edge) ignore the `timeout` completely.

Firefox does honor it but doesn't seem to apply any time offset. For a 5s timeout, it will reuse the connection 4.9 seconds later, but not 5.1 seconds.

I'll remove this offset.